queries.sql

```diff
@@ -39,6 +39,31 @@ SELECT AVG(weight_kg) FROM animals;
 SELECT neutered, MAX(escape_attempts) FROM animals GROUP BY neutered;
 SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;
 
+/* Vet clinic database: query multiple tables */
+SELECT name, full_name FROM animals a JOIN owners o ON o.id = a.owner_id WHERE full_name = 'Melody Pond';
+SELECT a.name, s.name FROM animals a JOIN species s ON s.id = a.species_id WHERE species_id = 1;
+SELECT o.*, name FROM owners o FULL JOIN animals a ON a.owner_id = o.id;
+SELECT species_id,s.name, COUNT(species_id) FROM animals a JOIN species s ON s.id = a.species_id GROUP BY species_id, s.name;
+SELECT full_name, a.name, s.name FROM animals a JOIN owners o ON a.owner_id = o.id JOIN species s ON s.id = a.species_id WHERE full_name = 'Jennifer Orwell' AND species_id = 2;
+SELECT full_name, a.name,escape_attempts FROM animals a JOIN owners o ON a.owner_id = o.id WHERE full_name=
+'Dean Winchester' AND escape_attempts > 0;
+SELECT full_name, MAX(owner_count) FROM (SELECT full_name, COUNT(owner_id) AS owner_count FROM animals a JOIN owners o ON a.owner_id = o.id GROUP BY full_name, owner_id) AS subquery GROUP BY full_name;
 
+/* Vet clinic database: add "join table" for visits */
 
-
+SELECT animals.name, vets.name, MAX(date_of_visit) AS last_vist FROM visits JOIN animals ON visits.animals_id = animals.id JOIN vets ON visits.vets_id = vets.id WHERE vets.name = 'William Tatcher' GROUP BY animals.name, vets.na
+me ORDER BY last_vist DESC LIMIT 1;
+SELECT animals.name, vets.name, COUNT(visits.animals_id) FROM visits JOIN animals ON visits.animals_id = animals.id JOIN vets ON vets.id = visits.vets_id WHERE vets.id = 3 GROUP BY animals.name, vets.name;
+SELECT vets.name, species.name, specializations.species_id FROM SPECIALIZATIONS FULL JOIN vets ON specializations.vets_id = vets.id FULL JOIN species ON species.id = specializations. species_id GROUP BY specializati
+ons.species_id, vets.name, species.name;
+SELECT animals.name AS animals_that_visited, visits.date_of_visit, vets.name FROM VISITS JOIN ANIMALS ON animals.id = visits.animals_id JOIN vets ON vets.id = visits.vets_id WHERE vets.id = 3 AND date_of_visit BETWEEN '2020-04-1' AND '2020-08-30' GROUP BY animals.name, vets.name, visits.date_of_visit;
+SELECT animals.name, COUNT(visits.animals_id) AS Animal_visits FROM visits JOIN animals ON animals.id = animals_id JOIN vets ON vets.id = visits.vets_id GROUP BY animals.name ORDER BY Animal_visits desc;
+SELECT animals.name, COUNT(visits.animals_id) AS Animal_visits FROM visits JOIN animals ON animals.id = animals_id JOIN vets ON vets.id = visits.vets_id GROUP BY animals.name ORDER BY Animal_visits;
+vet_clinic=# SELECT animals.name, COUNT(visits.animals_id) AS Animal_visits FROM visits JOIN animals ON animals.id = animals_id JOIN vets ON vets.id = visits.vets_id GROUP BY animals.name ORDER BY Animal_visits desc limit 2;
```

Comment:
- [ ] Remove all commented pieces of code from your documents to maintain a clean document.